test/CodeGen/AVR/sign-extension.ll

```diff
@@ -0,0 +1,11 @@
+; RUN: llc -march=avr < %s | FileCheck %s
+
+define i64 @sign_extended(i1) {
+; CHECK-LABEL: entry-block
+entry-block:
+  %1 = sext i1 %0 to i8
+  %2 = sext i8 %1 to i16
```

Comment:
> the whole 8-bit 16-bit weirdness that is going on.

Yeah, I think this is a killer here. After trying a few variations, I keep getting generated code that involves way too many shifts. I've a gut feeling that the "proper" fix is to untangle the 8-bit / 16-bit mess. I think I'll just add some comments to that effect and then merge this. The clever `sbc` code will still exist for reference when we can get around to it.